gst/droidcamsrc/gstdroidcamsrc.c

```diff
@@ -2633,3 +2661,37 @@ gst_droidcamsrc_post_preview (GstDroidCamSrc * src, GstSample * sample)
     gst_sample_unref (sample);
   }
 }
+
+static gboolean
+gst_droidcamsrc_update_jpeg_quality(GstDroidCamSrc * src)
+{
+  gboolean ret = FALSE;
+  const gchar* str = NULL;
+
+  GST_DEBUG_OBJECT (src, "update jpeg quality");
+
+  do {
+    g_rec_mutex_lock (&src->dev_lock);
+
+    if (!src->dev || !src->dev->params)
+      break;
+
+    str =
+      gst_droidcamsrc_params_get_string (src->dev->params, "jpeg-quality");
```

Comment:
Is there a reason for using string type here and below when the type of jpeg_quality is uint (should we even use uint? there is only gst_droidcamsrc_params_get_int), in some other parts of different types are used in similar functions.

Reply:
Shouldn't it be relatively easy to add error handling to the get_int?

Reply:
Was about to make it use strtol() myself and return -1, but is there actually any reason even. The atoi() is not undefined behavior, it returns 0 on error, and suppose we shouldn't allow 0 as quality anyway?

Reply:
Atoi() has undefined behavior if the value of the result cannot be represented. (See undefined behavior 119 of Annex J of the C Standard.) - https://wiki.sei.cmu.edu/confluence/display/c/CC.+Undefined+Behavior#CC.UndefinedBehavior-ub_119

More info for reference: https://wiki.sei.cmu.edu/confluence/display/c/ERR07-C.+Prefer+functions+that+support+error+checking+over+equivalent+functions+that+don%27t

Thats why it is not safe to use atoi() here. I agree with that. But I think that it should be part of another MR, 'cause this code could affect other parts of gst-droid.

Reply:
To me it seem quite clear how it behaves:
https://man7.org/linux/man-pages/man3/atoi.3.html

Those linked ones don't seem to be actual standard and didn't spot atoi() in C99 annex J.

Reply:
In real world it works like this to be honest: https://code.woboq.org/userspace/glibc/stdlib/atoi.c.html

But in theory it has UB: 

https://en.cppreference.com/w/cpp/string/byte/atoi
http://www.cplusplus.com/reference/cstdlib/atoi/

And static analylizer could possibly highlight it as UB.

So my suggestion is to keep current conversion as is or create separate MR and change atoi with strtol() before merging current MR.

Reply:
@pvuorela ok, lets replace it with get_int. I'll do that and retest in few days. Thanks for the review.

Reply:
@pvuorela Hello, I have fixed that + rebased on top of current master. Thanks!